paddle/fluid/framework/ir/mkldnn/params_quantization_mkldnn_pass_tester.cc

```diff
@@ -0,0 +1,305 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/framework/ir/mkldnn/params_quantization_mkldnn_pass.h"  // NOLINT
+#include <gtest/gtest.h>
+
+#include "paddle/fluid/imperative/type_defs.h"
+#include "paddle/fluid/platform/place.h"
+
+using LoDTensor = phi::DenseTensor;
+
+namespace paddle {
+namespace framework {
+namespace ir {
+namespace {
+struct Data {
+  Data() {}
+
+  Data(std::vector<int64_t>&& data_shape, std::vector<float>&& raw_data)
+      : shape(std::move(data_shape)), data(std::move(raw_data)) {
+    auto size_from_shape = std::accumulate(shape.begin(), shape.end(), 1,
+                                           std::multiplies<int64_t>());
+    PADDLE_ENFORCE_EQ(
+        size_from_shape, data.size(),
+        platform::errors::InvalidArgument("Shape and data don't match."));
+  }
+
+  const std::vector<int64_t>& getShape() const { return shape; }
+  const std::vector<float>& getData() const { return data; }
+
+ private:
+  const std::vector<int64_t> shape;
+  const std::vector<float> data;
+};
+
+struct TestScope {
+  void CreateTensor(const std::string& var_name, const Data& data) {
+    auto variable = scope.Var(var_name);
+    auto tensor = variable->GetMutable<LoDTensor>();
+    tensor->Resize(phi::make_ddim(data.getShape()));
+    auto dptr = tensor->mutable_data<float>(place);
+    std::copy(data.getData().begin(), data.getData().end(), dptr);
+  }
+
+  const LoDTensor& GetTensor(const std::string& input) const {
+    Variable* var = scope.FindVar(input);
+    return var->Get<LoDTensor>();
+  }
+
+  framework::Scope* Scope() { return &scope; }
+
+ private:
+  framework::Scope scope;
+  CPUPlace place;
+};
+
+struct ProgramStrategy {
+  virtual ~ProgramStrategy() {}
```

Comment:
Do you need this destructor here?

Reply:
Yes because I inherit from this base class in code below